backend/supabase/migrations/20240414161927_users.sql

```diff
@@ -0,0 +1,276 @@
+            SELECT
+                1
+            FROM
+                public.users u
+            WHERE
+                u.id = is_account_owner.user_id
+                AND u.account_name = is_account_owner.account_name)
+$$;
+
+GRANT EXECUTE ON FUNCTION app.is_account_owner(uuid, citext) TO authenticated, service_role;
+
+-------------------------------------------------------
+-- Section - Security
+-------------------------------------------------------
+ALTER TABLE public.users ENABLE ROW LEVEL SECURITY;
+
+GRANT UPDATE (display_name, bio, public_metadata) ON public.users TO authenticated;
+       
+CREATE POLICY users_delete_policy ON public.users AS permissive
+    FOR DELETE TO authenticated
+        USING (id = auth.uid());
+
+CREATE POLICY users_update_policy ON public.users AS permissive
+    FOR UPDATE TO authenticated
+        USING (id = auth.uid());
+
+CREATE POLICY users_select_policy ON public.users AS permissive
+    FOR SELECT TO authenticated
+        USING (TRUE);
+
+-------------------------------------------------------
+-- Section - Views
+-------------------------------------------------------
+
+-- TODO
+
+-------------------------------------------------------
+-- Section - RPC Functions
+-------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION public.get_user_id(account_name citext)
+    RETURNS uuid
+    SECURITY DEFINER
+    LANGUAGE plpgsql
+    STABLE
+    AS $$
+BEGIN
+    RETURN(
+        SELECT
+            u.id
+        FROM
+            users u
+        WHERE
+            u.account_name = get_user_id.account_name);
+END;
+$$;
+
+GRANT EXECUTE ON FUNCTION public.get_user_id(citext) TO authenticated, service_role;
+
+CREATE OR REPLACE FUNCTION public.get_user_by_id(user_id uuid)
+    RETURNS public.users
+    SET search_path = public
+    SECURITY DEFINER
+    LANGUAGE plpgsql
+    STABLE
+    AS $$
+BEGIN
+    RETURN u
+FROM
+    users u
+WHERE
+    u.id = get_user_by_id.user_id;
+END;
+$$;
+
+GRANT EXECUTE ON FUNCTION public.get_user_by_id(uuid) TO authenticated, service_role;
+
+CREATE OR REPLACE FUNCTION public.get_user_by_name(account_name citext)
+    RETURNS public.users
+    SET search_path = public
+    SECURITY DEFINER
+    LANGUAGE plpgsql
+    STABLE
+    AS $$
+BEGIN
+    RETURN u
+FROM
+    users u
+WHERE
+    u.account_name = get_user_by_name.account_name;
+END;
+$$;
+
+GRANT EXECUTE ON FUNCTION public.get_user_by_name(citext) TO authenticated, service_role;
+
+CREATE OR REPLACE FUNCTION public.get_me()
+    RETURNS public.users
+    SET search_path = public
+    SECURITY DEFINER
+    LANGUAGE plpgsql
+    STABLE
+    AS $$
+BEGIN
+    RETURN u
+FROM
+    users u
+WHERE
+    u.id = auth.uid();
+END;
+$$;
+
+GRANT EXECUTE ON FUNCTION public.get_me() TO authenticated;
+
+CREATE OR REPLACE FUNCTION public.update_user(user_id uuid, display_name text DEFAULT NULL, bio text DEFAULT NULL, public_metadata jsonb DEFAULT NULL, replace_metadata boolean DEFAULT FALSE)
```

Comment:
We'd probably want to restrict this access to only the authenticated user. Would this be automatically restricted by row level security?

Reply:
yes,
https://github.com/onlook-dev/onlook/pull/430/files#diff-a94914aeab25d03ab188b8d6032dce5c62168026eae485c707805d1d22143ea0R140-R142

Reply:
all functions have security of type `DEFINER` and all table RLS are enabled per row and column for insert/update/delete. however, these are needing more security check of course.